pkg/apiserver/rest/apis/v1/types.go

```diff
@@ -0,0 +1,250 @@
+	Name             string            `json:"name" validate:"required"`
+	Description      string            `json:"description,omitempty"`
+	Icon             string            `json:"icon"`
+	KubeConfig       string            `json:"kubeConfig" validate:"required_without=kubeConfigSecret"`
+	KubeConfigSecret string            `json:"kubeConfigSecret,omitempty" validate:"required_without=kubeConfig"`
+	Labels           map[string]string `json:"labels,omitempty"`
+}
+
+// DetailClusterResponse cluster detail information model
+type DetailClusterResponse struct {
+	ClusterBase
+	ResourceInfo ClusterResourceInfo `json:"resourceInfo"`
+	// remote manage url, eg. ACK cluster manage url.
+	RemoteManageURL string `json:"remoteManageURL,omitempty"`
+	// Dashboard URL
+	DashboardURL string `json:"dashboardURL,omitempty"`
+}
+
+// ClusterResourceInfo resource info of cluster
+type ClusterResourceInfo struct {
+	WorkerNumber     int      `json:"workerNumber"`
+	MasterNumber     int      `json:"masterNumber"`
+	MemoryCapacity   int64    `json:"memoryCapacity"`
+	CPUCapacity      int64    `json:"cpuCapacity"`
+	GPUCapacity      int64    `json:"gpuCapacity,omitempty"`
+	StorageClassList []string `json:"storageClassList,omitempty"`
+}
+
+// ListClusterResponse list cluster
+type ListClusterResponse struct {
+	Clusters []ClusterBase `json:"clusters"`
+}
+
+// ClusterBase cluster base model
+type ClusterBase struct {
+	Name        string            `json:"name"`
+	Description string            `json:"description"`
+	Icon        string            `json:"icon"`
+	Labels      map[string]string `json:"labels"`
+	Status      string            `json:"status"`
+	Reason      string            `json:"reason"`
+}
+
+// ListClusterAddonResponse list all addon of the cluster
+type ListClusterAddonResponse struct {
+	Addons []ClusterAddonBase `json:"addons"`
+}
+
+// ClusterAddonBase cluster addon base model
+type ClusterAddonBase struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Status      string `json:"status"`
+}
+
+// DeatilClusterAddonResponse detail addon info
+type DeatilClusterAddonResponse struct {
+	ClusterAddonBase
+}
+
+// ListApplicationResponse list applications by query params
+type ListApplicationResponse struct {
+	Applications []*ApplicationBase `json:"applications"`
+}
+
+// ApplicationBase application base model
+type ApplicationBase struct {
+	Name            string            `json:"name"`
+	Namespace       string            `json:"namespace"`
+	Description     string            `json:"description"`
+	CreateTime      time.Time         `json:"createTime"`
+	UpdateTime      time.Time         `json:"updateTime"`
+	Icon            string            `json:"icon"`
+	Labels          map[string]string `json:"labels,omitempty"`
+	ClusterBindList []ClusterBase     `json:"clusterList,omitempty"`
+	Status          string            `json:"status"`
+	GatewayRuleList []GatewayRule     `json:"gatewayRule"`
```

Comment:
can we use application policy to mark it? Why it's special？

Reply:
This data model is UI-oriented, that is, it contains application state data.